fluent/src/commonMain/kotlin/com/konyaco/fluent/component/Tree.kt

```diff
@@ -0,0 +1,299 @@
+        override val children: List<TreeElement<T>>,
+        override val onClick: ((Boolean) -> Unit)?
+    ) : TreeElement<T>()
+}
+
+fun <T> buildTree(block: TreeBuilder<T>.() -> Unit): Tree<T> {
+    val builder = TreeBuilder<T>()
+    builder.block()
+    return Tree(builder.buildRoots())
+}
+
+class TreeBuilder<T> {
+    private val elements = mutableListOf<BuilderElement<T>>()
+
+    fun node(data: T, id: Any = data.toString(), onClick: ((Boolean) -> Unit)? = null, children: NodeBuilder<T>.() -> Unit) {
+        val childBuilder = NodeBuilder<T>(0)
+        childBuilder.children()
+        elements.add(BuilderElement.Node(data, id, onClick, childBuilder.elements, 0))
+    }
+
+    fun leaf(data: T, id: Any = data.toString(), onClick: ((Boolean) -> Unit)? = null) {
+        elements.add(BuilderElement.Leaf(data, id, onClick, 0))
+    }
+
+    internal fun buildRoots(): List<TreeElement<T>> = elements.map { it.buildElement(depth = 0) }
+
+    sealed class BuilderElement<T> {
+        abstract fun buildElement(depth: Int): TreeElement<T>
+
+        data class Leaf<T>(val data: T, val id: Any, val onClick: ((Boolean) -> Unit)?, val depthInit: Int) : BuilderElement<T>() {
+            override fun buildElement(depth: Int): TreeElement<T> {
+                return TreeElement.Leaf(data, id, depth, onClick)
+            }
+        }
+
+        data class Node<T>(
+            val data: T,
+            val id: Any,
+            val onClick: ((Boolean) -> Unit)?,
+            val children: List<BuilderElement<T>>,
+            val depthInit: Int
+        ) : BuilderElement<T>() {
+            override fun buildElement(depth: Int): TreeElement<T> {
+                val builtChildren = children.map { it.buildElement(depth + 1) }
+                return TreeElement.Node(data, id, depth, builtChildren, onClick)
+            }
+        }
+    }
+}
+
+class NodeBuilder<T>(private val parentDepth: Int) {
+    internal val elements = mutableListOf<TreeBuilder.BuilderElement<T>>()
+
+    fun node(data: T, id: Any = data.toString(), onClick: ((Boolean) -> Unit)? = null, children: NodeBuilder<T>.() -> Unit) {
+        val childBuilder = NodeBuilder<T>(parentDepth + 1)
+        childBuilder.children()
+        elements.add(TreeBuilder.BuilderElement.Node(data, id, onClick, childBuilder.elements, parentDepth + 1))
+    }
+
+    fun leaf(data: T, id: Any = data.toString(), onClick: ((Boolean) -> Unit)? = null) {
+        elements.add(TreeBuilder.BuilderElement.Leaf(data, id, onClick, parentDepth + 1))
+    }
+}
+
+// -------------------------------------
+// Tree Node Color Scheme
+// -------------------------------------
+
+@Immutable
+data class TreeNodeColor(
+    val backgroundColor: Color,
+    val contentColor: Color,
+    val borderColor: Color,
+    val labelTextColor: Color
+)
+
+typealias TreeNodeColorScheme = PentaVisualScheme<TreeNodeColor>
+
+object TreeNodeDefaults {
+    @Stable
+    @Composable
+    fun defaultNodeColors(
```

Comment:
default color token should reference figma file, see https://www.figma.com/community/file/1440832812269040007/windows-ui-kit